algorithms/src/traits/snark.rs

```diff
@@ -20,14 +20,21 @@ use snarkvm_utilities::{CanonicalDeserialize, CanonicalSerialize, FromBytes, ToB
 use rand::{CryptoRng, Rng};
 use snarkvm_fields::{PrimeField, ToConstraintField};
 use snarkvm_r1cs::ConstraintSynthesizer;
-use std::{borrow::Borrow, fmt::Debug, sync::atomic::AtomicBool};
+use std::{borrow::Borrow, collections::BTreeMap, fmt::Debug, sync::atomic::AtomicBool};
 
-/// Defines a trait that describes preparing from an unprepared version to a prepare version.
+/// Defines trait that describes preparing from an unprepared version to a prepare version.
 pub trait Prepare {
     type Prepared;
     fn prepare(&self) -> Self::Prepared;
 }
 
+/// Defines trait that describes preparing from an unprepared version to an orderable prepare version.
+pub trait PrepareOrd {
+    // NOTE: merging this with the Prepare trait is possible but introduces more complexity
+    type Prepared: Ord;
+    fn prepare(&self) -> Self::Prepared;
+}
```

Comment:
Can you elaborate why "merging this with the `Prepare` trait [...] introduces more complexity"?

Reply:
See comment above, updated it to `    // NOTE: we keep this separate from Prepare because we also have unordered Prepared types`